pkg/certificateprovider/azurekeyvault/provider.go

```diff
@@ -66,34 +66,34 @@ func (s *akvCertProvider) GetCertificates(ctx context.Context, attrib map[string
 	workloadIdentityClientID := types.GetClientID(attrib)
 
 	if keyvaultURI == "" {
-		return nil, nil, fmt.Errorf("keyvaultUri is not set")
+		return nil, nil, re.ErrorCodeConfigInvalid.NewError(re.CertProvider, providerName, re.AKVLink, nil, "keyvaultUri is not set", false)
 	}
 	if tenantID == "" {
-		return nil, nil, fmt.Errorf("tenantID is not set")
+		return nil, nil, re.ErrorCodeConfigInvalid.NewError(re.CertProvider, providerName, re.AKVLink, nil, "tenantID is not set", false)
 	}
 	if workloadIdentityClientID == "" {
-		return nil, nil, fmt.Errorf("clientID is not set")
+		return nil, nil, re.ErrorCodeConfigInvalid.NewError(re.CertProvider, providerName, re.AKVLink, nil, "clientID is not set", false)
 	}
 
 	azureCloudEnv, err := parseAzureEnvironment(cloudName)
 	if err != nil {
-		return nil, nil, fmt.Errorf("cloudName %s is not valid, error: %w", cloudName, err)
+		return nil, nil, re.ErrorCodeConfigInvalid.NewError(re.CertProvider, providerName, "", nil, fmt.Sprintf("cloudName %s is not valid", cloudName), false)
 	}
 
 	keyVaultCerts, err := getKeyvaultRequestObj(attrib)
 	if err != nil {
-		return nil, nil, fmt.Errorf("failed to get keyvault request object from provider attributes, error: %w", err)
+		return nil, nil, re.ErrorCodeConfigInvalid.NewError(re.CertProvider, providerName, re.AKVLink, err, "failed to get keyvault request object from provider attributes", false)
 	}
 
 	if len(keyVaultCerts) == 0 {
-		return nil, nil, fmt.Errorf("no keyvault certificate configured")
+		return nil, nil, re.ErrorCodeConfigInvalid.NewError(re.CertProvider, providerName, "", nil, "no keyvault certificate configured", true)
```

Comment:
curious what is the the "true" , "false" value represent, wondering if it make sense to have a named variable here.

Reply:
Added constants for them